### ISSUE TITLE
TST: quantity ufunc has 4 failures in numpy 2.0.0.dev (but really it  is 1.26.dev)

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#14970]
Numpy: 2.0.0.dev0+279.gdbc454a15

astropy/units/tests/test_quantity_non_ufuncs.py

```
__________________________ test_testing_completeness ___________________________

    @needs_array_function
    def test_testing_completeness():
        assert not CoverageMeta.covered.intersection(untested_functions)
>       assert all_wrapped == (CoverageMeta.covered | untested_functions)
E       assert {<function ar...a758130>, ...} == {<function ar...6d6e130>, ...}
E         Extra items in the left set:
E         <function byte_bounds at 0x7f3576d9aca0>
E         <function who at 0x7f3576d9ad40>
E         <function safe_eval at 0x7f3576d9b420>
E         Full diff:
E           {
E         +  <function byte_bounds at 0x7f3576d9aca0>,
E         +  <function who at 0x7f3576d9ad40>,
E         +  <function safe_eval at 0x7f3576d9b420>,
E            <function merge_arrays at 0x7f35746891f0>,
E            <function drop_fields at 0x7f35746898b0>,
...
```

```
______________ TestFunctionHelpersCompleteness.test_all_included _______________

self = <astropy.units.tests.test_quantity_non_ufuncs.TestFunctionHelpersCompleteness object at ...>

    @needs_array_function
    def test_all_included(self):
        included_in_helpers = (
            SUBCLASS_SAFE_FUNCTIONS
            | UNSUPPORTED_FUNCTIONS
            | set(FUNCTION_HELPERS.keys())
            | set(DISPATCHED_FUNCTIONS.keys())
        )
>       assert all_wrapped == included_in_helpers
E       assert {<function ar...a758130>, ...} == {<function ar...6d48170>, ...}
E         Extra items in the left set:
E         <function byte_bounds at 0x7f3576d9aca0>
E         <function who at 0x7f3576d9ad40>
E         <function safe_eval at 0x7f3576d9b420>
E         Full diff:
E           {
E         +  <function byte_bounds at 0x7f3576d9aca0>,
E         +  <function who at 0x7f3576d9ad40>,
E         +  <function safe_eval at 0x7f3576d9b420>,
E            <function merge_arrays at 0x7f35746891f0>,
E            <function drop_fields at 0x7f35746898b0>,
...
```

astropy/utils/masked/tests/test_function_helpers.py

```
_______________________ test_basic_testing_completeness ________________________

    def test_basic_testing_completeness():
>       assert all_wrapped == (tested_functions | IGNORED_FUNCTIONS | UNSUPPORTED_FUNCTIONS)
E       assert {<function ar...6da0130>, ...} == {<function ar...a758130>, ...}
E         Extra items in the left set:
E         <function byte_bounds at 0x7f3576d9aca0>
E         <function who at 0x7f3576d9ad40>
E         <function safe_eval at 0x7f3576d9b420>
E         Full diff:
E           {
E         +  <function byte_bounds at 0x7f3576d9aca0>,
E         +  <function who at 0x7f3576d9ad40>,
E         +  <function safe_eval at 0x7f3576d9b420>,
E            <function broadcast_to at 0x7f3576d145b0>,
E            <function broadcast_arrays at 0x7f3576d147f0>,
...
```

```
______________ TestFunctionHelpersCompleteness.test_all_included _______________

self = <astropy.utils.masked.tests.test_function_helpers.TestFunctionHelpersCompleteness object at ...>

    def test_all_included(self):
        included_in_helpers = (
            MASKED_SAFE_FUNCTIONS
            | UNSUPPORTED_FUNCTIONS
            | set(APPLY_TO_BOTH_FUNCTIONS.keys())
            | set(DISPATCHED_FUNCTIONS.keys())
        )
>       assert all_wrapped == included_in_helpers
E       assert {<function ar...6da0130>, ...} == {<function ar...6d6e130>, ...}
E         Extra items in the left set:
E         <function byte_bounds at 0x7f3576d9aca0>
E         <function who at 0x7f3576d9ad40>
E         <function safe_eval at 0x7f3576d9b420>
E         Full diff:
E           {
E         +  <function byte_bounds at 0x7f3576d9aca0>,
E         +  <function who at 0x7f3576d9ad40>,
E         +  <function safe_eval at 0x7f3576d9b420>,
E            <function broadcast_to at 0x7f3576d145b0>,
E            <function broadcast_arrays at 0x7f3576d147f0>,
...
```

**Comment by pllim** [comment]
Looks like these 3 functions were deprecated in https://github.com/numpy/numpy/pull/23830 but I dunno where to insert the code in `astropy` to make the tests happy. They will land in numpy 1.26.0 according to https://github.com/numpy/numpy/pull/23830#issuecomment-1591709865 .

* byte_bounds
* who
* safe_eval

**Comment by mhvk** [comment]
fix in #14973
